vars/uploadMinSnapshotsToS3.groovy

```diff
@@ -1,15 +1,41 @@
 void call(Map args = [:]) {
+
     def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+    List<Closure> fileActions = args.fileActions ?: []
     String manifest = args.manifest ?: "manifests/${INPUT_MANIFEST}"
+
     def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: manifest))
     echo("Retreving build manifest from: $WORKSPACE/builds/${inputManifest.build.getFilename()}/manifest.yml")
-    def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: "$WORKSPACE/builds/${inputManifest.build.getFilename()}/manifest.yml"))
+
+    productName = inputManifest.build.getFilename()
+    def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: "$WORKSPACE/builds/${productName}/manifest.yml"))
     version = buildManifest.build.version
     architecture = buildManifest.build.architecture
     platform = buildManifest.build.platform
-    minArtifactSourceName = "opensearch-min-${version}-${platform}-${architecture}.tar.gz"
-    minArtifactName = "opensearch-min-${version}-${platform}-${architecture}-latest.tar.gz"
+    id = buildManifest.build.id
+
+    // Setup src & dst variables for artifacts
+    srcDir = "${WORKSPACE}/builds/${productName}/dist"
+    dstDir = "snapshots/core/${productName}/${version}"
+    baseName = "${productName}-min-${version}-${platform}-${architecture}"
+
+    // Create checksums
+    echo("Create .sha512 for Min Snapshots Artifacts")
+    argsMap = [:]
+    argsMap['artifactPath'] = srcDir
+    for (Closure action : fileActions) { // running createSha512Checksums()
+        action(argsMap)
+    }
+
     withAWS(role: "${ARTIFACT_PROMOTION_ROLE_NAME}", roleAccount: "${AWS_ACCOUNT_ARTIFACT}", duration: 900, roleSessionName: 'jenkins-session') {
-        s3Upload(file: "$WORKSPACE/builds/opensearch/dist/${minArtifactSourceName}", bucket: "${ARTIFACT_PRODUCTION_BUCKET_NAME}", path: "snapshots/core/opensearch/${version}/${minArtifactName}")
+        sh """
+            cp ${srcDir}/${baseName}.tar.gz.sha512 ${srcDir}/${baseName}-${id}.tar.gz.sha512 ${srcDir}/${baseName}-latest.tar.gz.sha512
+            sed -i "s/.tar.gz.sha512/-${id}.tar.gz.sha512/g" ${srcDir}/${baseName}-${id}.tar.gz.sha512
```

Comment:
We do not need id in snapshots name anywhere

Reply:
Sorry I was thinking about release-candidates and not snapshots. But even for snapshots do we need build number included in the name? I don't think so.
cc: @dblock

Reply:
I think we still need it @peternied can you help confirm?
Adding id copy would help us know whether we have successfully uploaded or not, even retain an older copy of it.
Thanks.

Reply:
It defeats the purpose of having snapshots doesn't it? If it is not successful to upload the job will fail and we would know then. Why keep a copy when you are already uploading the `-latest`? We have a copy in staging bucket.

Reply:
I don't think its useful to keep around the 'id' artifacts, I would prefer that we consult the log for the build job if we need to confirm behavior

Reply:
Updated now.